components/tidb_query_datatype/src/codec/table.rs

```diff
@@ -199,14 +200,29 @@ pub fn decode_int_handle(mut key: &[u8]) -> Result<i64> {
     key.read_i64().map_err(Error::from)
 }
 
-/// `decode_common_handle` decodes key key and gets the common handle.
+/// `decode_common_handle` decodes key and gets the common handle.
 #[inline]
 pub fn decode_common_handle(mut key: &[u8]) -> Result<&[u8]> {
     check_record_key(key)?;
     key = &key[PREFIX_LEN..];
     Ok(key)
 }
 
+/// `decode_common_handle_into_datums` decodes encoded common handle into a datum array.
+#[inline]
+pub fn decode_common_handle_into_datums(mut handle: &[u8]) -> Result<Vec<Datum>> {
+    datum::decode(&mut handle)
+}
+
+/// `encode_common_handle_from_datums` encodes datum array into a encoded common handle.
+#[inline]
+pub fn encode_common_handle_from_datums(
+    ctx: &mut EvalContext,
+    values: &[Datum],
+) -> Result<Vec<u8>> {
+    datum::encode_key(ctx, values)
```

Comment:
How about `datum::encode_value` instead, `datum::encode_key` will encode a memory comparable buffer, which is necessary since it is just a temporary format.

Reply:
This procedure is migrated from `tidb:util/testutil/testutil.go:121`:
```go
// MustNewCommonHandle create a common handle with given values.
// TODO: please use testkit.MustNewCommonHandle to replace this function to migrate to testify
func MustNewCommonHandle(c *check.C, values ...interface{}) kv.Handle {
	encoded, err := codec.EncodeKey(new(stmtctx.StatementContext), nil, types.MakeDatums(values...)...)
	c.Assert(err, check.IsNil)
	ch, err := kv.NewCommonHandle(encoded)
	c.Assert(err, check.IsNil)
	return ch
}
```
However I've not tested it with real common handle data yet. We may reconsider it later.